homeassistant/components/moehlenhoff_alpha2/config_flow.py

```diff
@@ -0,0 +1,59 @@
+"""Alpha2 config flow."""
+import asyncio
+import logging
+
+import aiohttp
+from moehlenhoff_alpha2 import Alpha2Base
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({"host": str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    base = Alpha2Base(data["host"])
+    await base.update_data()
+
+    # Return info that you want to store in the config entry.
+    return {"title": base.name}
+
+
+class Alpha2BaseConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Möhlenhoff Alpha2 config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            try:
```

Comment:
please wrap this around the least amount of code as possible, maybe even move the try check into the validate function and then return the errors and/or title.

Reply:
Moved the handling of ClientConnectorError and TimeoutError to validate_input().

Reply:
There is still too much code in the try block.
Move `except Exception` to validate_input and remove the try here.

Reply:
Catch `Exception` in `validate_input` so this try can be removed.

Reply:
Done.